loader/icd.c

```diff
@@ -26,13 +26,16 @@
 #include <string.h>
 
 KHRicdVendor *khrIcdVendors = NULL;
+char *khrEnableTrace = NULL;
+
 #if defined(CL_ENABLE_LAYERS)
 struct KHRLayer *khrFirstLayer = NULL;
 #endif // defined(CL_ENABLE_LAYERS)
 
 // entrypoint to initialize the ICD and add all vendors
 void khrIcdInitialize(void)
 {
+    khrEnableTrace = khrIcd_getenv("OCL_ICD_ENABLE_TRACE");
```

Comment:
Two remarks here

- Same remark as in the issue, discussion of using `khrIcd_getenv` vs `khrIcd_secure_getenv`. Tagging @jenatali as the windows implications seem stronger than the Linux ones (I may be wrong).
- the initialization of the tracing variable should go in it's own subroutine, maybe per os, depending on the result of the above discussion. This initialization should end up with `khrEnableTrace` containing a boolean value signifying if the tracing is active or not. The list of acceptable values for the string will need to be determined but will most probably end being more than just `"True"`

Reply:
For item 1,  I will wait for @jenatali 's feedback.  Personally I still think this env variable does not change the execution logic of loading drivers. So we do not need the khrIcd_secure_getenv.

For item 2, I added "true", "T", "1".   I added the initialization of trace enabling because khrIcdInitialize is the entry point to initialize ICD.